Specs/Core/EllipsoidRhumbLineSpec.js

```diff
@@ -0,0 +1,581 @@
+        var end = new Cartographic(CesiumMath.PI_OVER_FOUR, 0.0);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(CesiumMath.PI_OVER_TWO * 6).toEqualEpsilon(rhumb.surfaceDistance, CesiumMath.EPSILON12);
+    });
+
+    it('computes distance at meridian', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 6);
+        var fifteenDegrees = Math.PI / 12;
+        var start = new Cartographic(CesiumMath.PI_OVER_TWO, fifteenDegrees);
+        var fortyfiveDegrees = Math.PI / 4;
+        var end = new Cartographic(CesiumMath.PI_OVER_TWO, fortyfiveDegrees);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        var thirtyDegrees = Math.PI / 6;
+        expect(thirtyDegrees * 6).toEqualEpsilon(rhumb.surfaceDistance, CesiumMath.EPSILON12);
+    });
+
+    it('computes distance for sphere 90 degrees along meridian and equator and check equality', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 6);
+        var fortyFiveSouth = new Cartographic(0.0, -CesiumMath.PI_OVER_FOUR);
+        var fortyFiveNorth = new Cartographic(0.0, CesiumMath.PI_OVER_FOUR);
+        var fortyFiveWest = new Cartographic(-CesiumMath.PI_OVER_FOUR, 0.0);
+        var fortyFiveEast = new Cartographic(CesiumMath.PI_OVER_FOUR, 0.0);
+
+        var westEastRhumb = new EllipsoidRhumbLine(fortyFiveWest, fortyFiveEast, ellipsoid);
+        var southNorthRhumb = new EllipsoidRhumbLine(fortyFiveSouth, fortyFiveNorth, ellipsoid);
+        var eastWestRhumb = new EllipsoidRhumbLine(fortyFiveEast, fortyFiveWest, ellipsoid);
+        var northSouthRhumb = new EllipsoidRhumbLine(fortyFiveNorth, fortyFiveSouth, ellipsoid);
+        expect(CesiumMath.PI_OVER_TWO * 6).toEqualEpsilon(westEastRhumb.surfaceDistance, CesiumMath.EPSILON12);
+        expect(CesiumMath.PI_OVER_TWO * 6).toEqualEpsilon(southNorthRhumb.surfaceDistance, CesiumMath.EPSILON12);
+        expect(westEastRhumb.surfaceDistance).toEqualEpsilon(southNorthRhumb.surfaceDistance, CesiumMath.EPSILON12);
+
+        expect(CesiumMath.PI_OVER_TWO * 6).toEqualEpsilon(eastWestRhumb.surfaceDistance, CesiumMath.EPSILON12);
+        expect(CesiumMath.PI_OVER_TWO * 6).toEqualEpsilon(northSouthRhumb.surfaceDistance, CesiumMath.EPSILON12);
+        expect(eastWestRhumb.surfaceDistance).toEqualEpsilon(northSouthRhumb.surfaceDistance, CesiumMath.EPSILON12);
+    });
+
+    it('computes distance at same latitude', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 6);
+        var fortyfiveDegrees = Math.PI / 4;
+        var start = new Cartographic(0, -fortyfiveDegrees);
+        var end = new Cartographic(CesiumMath.PI_OVER_TWO, -fortyfiveDegrees);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        var distance = Math.cos(fortyfiveDegrees) * CesiumMath.PI_OVER_TWO * 6;
+        expect(distance).toEqualEpsilon(rhumb.surfaceDistance, CesiumMath.EPSILON12);
+    });
+
+    it('tests sphere', function() {
```

Comment:
Maybe computes distance and heading given endpoints on a sphere?
Is it also a little suspect that the spec checks distance and heading against results computed in EllipsoidRhumb?

Reply:
It is to ensure that both ways gives the correct result. I'll change the heading.